brave/brave-ios
# Fix #4435: Add Privacy Reports feature.

## Summary of Changes


This pull request fixes #4435 
Security review: https://github.com/brave/security/issues/844

## Submitter Checklist:


## Test Plan:

Look for `Privacy Hub iOS Product Overview` spec in google drive for product overview and tips for testing.

This is a new feature and should be tested as such. It is supposed to show items blocked either by Brave Shields or the Brave VPN alerts.
Few testing scenarios:
Open few ad heavy websites. Wait a a minute, then open the privacy reports screen, see if the website you visited are tracked there. See if tracker names look correct.
Go to the 'All time lists' section, see if websites are there if tracker are present, some tracker might have an entity name under them.

To test the VPN Alert you have to buy a VPN product, can be done for free with Testflight. Can be done on production only, on Dev builds there's dummy vpn data pushed every 1 minute.

Test app upgrade 1.37 -> 1.38 with the privacy hub feature.

Test onboarding:
- Fresh install 1.38, do not open privacy reports screen.
- Open few websites, on dev 20 trackers must be detected, on prod 250 before the onboarding shows up.
- After tracker limit threshold is hit, open new tab page.
- There should be a popup pointing to the shield stats view, verify it's there and tapping on the button opens the privacy hub.

Test notification: dev build only
- Enable Privacy Hub notifications
- Background the app wait above 5 minutes
- Notification should show up
- Tap on it and see if it takes you to the Privacy Hub screen
On prod this notification is shown every Sunday at 11 AM

Settings test:
Two toggles to check out, it's in Brave Shields -> Privacy Hub setting:
- Disable Shields toggle, verify no new data is saved to the privacy hub
- Disable vpn alerts, verify no new vpn alerts data is saved, the section should be not visible anymore.

Privacy tests:
- Make sure no Brave Shields data is logged when browsing in private mode
- Make sure Privacy Hub can't be tapped in privacy mode to be opened.

## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
